nemo/collections/asr/modules/hat.py

```diff
@@ -0,0 +1,531 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import torch
+from omegaconf import DictConfig, OmegaConf
+
+from nemo.collections.asr.modules import rnnt, rnnt_abstract
+from nemo.collections.asr.parts.submodules import stateless_net
+from nemo.collections.asr.parts.utils import adapter_utils, rnnt_utils
+from nemo.collections.common.parts import rnn
+from nemo.core.classes import adapter_mixins, typecheck
+from nemo.core.classes.exportable import Exportable
+from nemo.core.classes.mixins import AdapterModuleMixin
+from nemo.core.neural_types import (
+    AcousticEncodedRepresentation,
+    ElementType,
+    EmbeddedTextType,
+    LabelsType,
+    LengthsType,
+    LogprobsType,
+    LossType,
+    NeuralType,
+    SpectrogramType,
+)
+from nemo.utils import logging
+
+
+class HATJoint(rnnt_abstract.AbstractRNNTJoint, Exportable, AdapterModuleMixin):
```

Comment:
This class is duplicating a lot of code from RNNTJoint. Would it make sense to subclass it ?